osu.Game/Skinning/Components/BeatmapAttributeText.cs

```diff
@@ -74,34 +92,38 @@ protected override void LoadComplete()
+            foreach (var attribute in Enum.GetValues<BeatmapAttribute>())
+                valueDictionary[attribute] = string.Empty;
+
+            ruleset.BindValueChanged(_ => updateAll());
+
             Template.BindValueChanged(_ => updateLabel());
-            beatmap.BindValueChanged(b =>
+            Attribute.BindValueChanged(_ =>
             {
-                updateBeatmapContent(b.NewValue);
-                updateLabel();
-            }, true);
-        }
+                updateBindableDifficulty();
+                updateAll();
+            });
 
-        private void updateBeatmapContent(WorkingBeatmap workingBeatmap)
-        {
-            valueDictionary[BeatmapAttribute.Title] = new RomanisableString(workingBeatmap.BeatmapInfo.Metadata.TitleUnicode, workingBeatmap.BeatmapInfo.Metadata.Title);
-            valueDictionary[BeatmapAttribute.Artist] = new RomanisableString(workingBeatmap.BeatmapInfo.Metadata.ArtistUnicode, workingBeatmap.BeatmapInfo.Metadata.Artist);
-            valueDictionary[BeatmapAttribute.DifficultyName] = workingBeatmap.BeatmapInfo.DifficultyName;
-            valueDictionary[BeatmapAttribute.Creator] = workingBeatmap.BeatmapInfo.Metadata.Author.Username;
-            valueDictionary[BeatmapAttribute.Source] = workingBeatmap.BeatmapInfo.Metadata.Source;
-            valueDictionary[BeatmapAttribute.Length] = TimeSpan.FromMilliseconds(workingBeatmap.BeatmapInfo.Length).ToFormattedDuration();
-            valueDictionary[BeatmapAttribute.RankedStatus] = workingBeatmap.BeatmapInfo.Status.GetLocalisableDescription();
-            valueDictionary[BeatmapAttribute.BPM] = workingBeatmap.BeatmapInfo.BPM.ToLocalisableString(@"F2");
-            valueDictionary[BeatmapAttribute.CircleSize] = ((double)workingBeatmap.BeatmapInfo.Difficulty.CircleSize).ToLocalisableString(@"F2");
-            valueDictionary[BeatmapAttribute.HPDrain] = ((double)workingBeatmap.BeatmapInfo.Difficulty.DrainRate).ToLocalisableString(@"F2");
-            valueDictionary[BeatmapAttribute.Accuracy] = ((double)workingBeatmap.BeatmapInfo.Difficulty.OverallDifficulty).ToLocalisableString(@"F2");
-            valueDictionary[BeatmapAttribute.ApproachRate] = ((double)workingBeatmap.BeatmapInfo.Difficulty.ApproachRate).ToLocalisableString(@"F2");
-            valueDictionary[BeatmapAttribute.StarRating] = workingBeatmap.BeatmapInfo.StarRating.ToLocalisableString(@"F2");
+            beatmap.BindValueChanged(_ =>
+            {
+                updateBindableDifficulty();
+                updateAll();
+            });
+
+            mods.BindValueChanged(_ =>
+            {
+                modSettingChangeTracker?.Dispose();
+                modSettingChangeTracker = new ModSettingChangeTracker(mods.Value);
+                modSettingChangeTracker.SettingChanged += _ => updateAll();
+                updateAll();
+            }, true);
         }
 
         private void updateLabel()
         {
+            if (IsLoaded == false)
```

Comment:
```suggestion
            if (!IsLoaded)
```

Also why is this only here and not in any of the other methods? And does this actually happen in practice?

Reply:
there was a case when function `updateLabel` was called before Drawable was loaded
I don't sure is this still possible or no, I will test it